### ISSUE TITLE
Dotted notation for declaring class methods

### ISSUE THREAD

**Issue by rtpg** [rtpg/kwil#23]
This is a way I figured out of declaring methods without too much effort on the transpiling side

```
data List = Cons x xs | Nil;

List::append = \a b -> case a of 
    Nil -> b
  | Cons x xs -> Cons x (xs.append b);
```

We'll end up putting these onto the prototypes of the `List` objects and having `Cons` and `Nil` extend from those. But we'll also want `append` to be available by itself, and for dispatch to happen on it. So you could define `append` for `List` but also for `Set` or something if you wanted to.

**Comment by rtpg** [comment]
Insetad we're going to go with a `::` notation. This will let me easily detect this.

How this will work is that this will get put as ```
List.prototype.append = function(arg){ return function(a){ return function(b){ ... }}(this)(arg)}
```

The double application will give us python-like "self" behaviour

**Comment by rtpg** [comment]
This doesn't support constant values for now... we'll work on this later

**Comment by rtpg** [comment]
implemented in #25
